compiler/AST/build.cpp

```diff
@@ -2102,7 +2102,13 @@ BlockStmt* convertTypesToExtern(BlockStmt* blk) {
       if (!de->init) {
         Symbol* vs = de->sym;
         PrimitiveType* pt = new PrimitiveType(NULL);
-        DefExpr* newde = new DefExpr(new TypeSymbol(vs->name, pt));
+
+        TypeSymbol* ts = new TypeSymbol(vs->name, pt);
+        if (VarSymbol* theVs = toVarSymbol(vs)) {
+          ts->doc = theVs->doc;
+        }
```

Comment:
It looks like the no doc flag is preserved. At least this test case works without changing convertTypeToExtern --> https://github.com/thomasvandoren/chapel/commit/2db2647d8cb4c4a4feb8bed2f6ebc7fc71f56503#diff-3830bfd6681c61d4f7aeae4599748530R25